### PR TITLE
Add TLS support to http server

### PR DESCRIPTION
**Pull request by danfaizer**
This PR provides native TLS support. Related issue: #297 

In order to enable TLS you must configure:
```
vouch:
  tls:
    cert: /path/to/signed_cert_plus_intermediates
    key: /path/to/private_key
    # profile: Optional. Define TLS configuration:  modern, intermediate, old, default
    profile: intermediate
```

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -166,6 +166,11 @@ func main() {
 		ErrorLog:     log.New(&fwdToZapWriter{fastlog}, "", 0),
 	}
 
+	if cfg.Cfg.TLS.Cert != "" || cfg.Cfg.TLS.Key != "" {
+		srv.TLSConfig = cfg.TLSConfig(cfg.Cfg.TLS.Profile)
+		logger.Fatal(srv.ListenAndServeTLS(cfg.Cfg.TLS.Cert, cfg.Cfg.TLS.Key))
+	}
+
```
Comment (bnfinet) [inline main.go:173]: which `logger.Fatal` is an exit event could you put the following `srv.ListenAndServe` (non TLS) in an `else` block for clarity

File: config/config.yml_example [hunk 2]
```diff
@@ -55,6 +55,11 @@ vouch:
   # - myOrg
   # - myOrg/myTeam
 
+  tls:
```
Comment (bnfinet) [inline config/config.yml_example:58]: Please include the the env var names in the documentation

**Comment by danfaizer** [comment]
Including coverage:

```
./do.sh coverage
?   	github.com/danfaizer/vouch-proxy	[no test files]
ok  	github.com/danfaizer/vouch-proxy/handlers	5.728s	coverage: 71.1% of statements
?   	github.com/danfaizer/vouch-proxy/pkg/capturewriter	[no test files]
ok  	github.com/danfaizer/vouch-proxy/pkg/cfg	0.241s	coverage: 59.5% of statements
ok  	github.com/danfaizer/vouch-proxy/pkg/cookie	(cached)	coverage: 42.3% of statements
ok  	github.com/danfaizer/vouch-proxy/pkg/domains	(cached)	coverage: 79.2% of statements
?   	github.com/danfaizer/vouch-proxy/pkg/healthcheck	[no test files]
ok  	github.com/danfaizer/vouch-proxy/pkg/jwtmanager	0.296s	coverage: 46.1% of statements
?   	github.com/danfaizer/vouch-proxy/pkg/providers/adfs	[no test files]
?   	github.com/danfaizer/vouch-proxy/pkg/providers/azure	[no test files]
?   	github.com/danfaizer/vouch-proxy/pkg/providers/common	[no test files]
ok  	github.com/danfaizer/vouch-proxy/pkg/providers/github	0.478s	coverage: 72.3% of statements
?   	github.com/danfaizer/vouch-proxy/pkg/providers/google	[no test files]
?   	github.com/danfaizer/vouch-proxy/pkg/providers/homeassistant	[no test files]
?   	github.com/danfaizer/vouch-proxy/pkg/providers/indieauth	[no test files]
?   	github.com/danfaizer/vouch-proxy/pkg/providers/nextcloud	[no test files]
?   	github.com/danfaizer/vouch-proxy/pkg/providers/openid	[no test files]
?   	github.com/danfaizer/vouch-proxy/pkg/providers/openstax	[no test files]
?   	github.com/danfaizer/vouch-proxy/pkg/responses	[no test files]
?   	github.com/danfaizer/vouch-proxy/pkg/structs	[no test files]
?   	github.com/danfaizer/vouch-proxy/pkg/timelog	[no test files]
```

File: main.go (hunk 1 above)
Comment (danfaizer) [reply main.go:181]: ✅

File: config/config.yml_example (hunk 2 above)
Comment (danfaizer) [reply config/config.yml_example:58]: ✅

**Pull request merged**